### DISCUSSION TITLE
Trainer Args from YAML file?

### DISCUSSION THREAD

**Discussion by FeryET** [Lightning-AI/pytorch-lightning#10721] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi.

I'm trying to refactor my code, and one of the things I would like to do is to move most of my configurations to a YAML file, which supports a way better hierarchical configuration pattern, and is more readable and more configurable. I can write a manual interface for parsing the YAML file and creating a Trainer instance with it, but I was wondering is there any better way to do this?

**Comment by tchaton** [comment]
Dear @FeryET.

I would advise you to use the LightningCLI, it would generate automatically the YAML file for you.
Check out the associated documentation: https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.utilities.cli.html?highlight=LightningCLI#pytorch_lightning.utilities.cli.LightningCLI
